setup.py

```diff
@@ -187,6 +187,22 @@ def run(self) -> None:
             self.move_output(ext)
 
     def move_output(self, ext: CMakeExtension) -> None:
+        if ext.name == "lcmtypes":
+            build_temp_path = Path(self.build_temp)
+            dest_path_dir = Path(self.get_ext_fullpath(ext.name)).resolve().parent  # type: ignore[attr-defined]
+            if self.inplace:
+                # NOTE(brad): If building in-place, place package in lcmtypes_build dir to avoid
+                # collision with existing lcmtypes directory. Happens in editable installs.
+                dest_path = dest_path_dir / "lcmtypes_build" / "lcmtypes"
```

Comment:
Should we just always do this?

Reply:
We probably can't just always do this - for non-editable installs, this needs to put these files in the actual expected final location (within a temporary tree that will get copied into the actual install location)